vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/command/PrepareStatementCommand.java

```diff
@@ -22,18 +22,25 @@
 public class PrepareStatementCommand extends CommandBase<PreparedStatement> {
 
   private final String sql;
-  public long statement; // 0 means unamed statement otherwise CString
+  private final boolean auto;
 
-  public PrepareStatementCommand(String sql) {
+  public PrepareStatementCommand(String sql, boolean auto) {
     this.sql = sql;
+    this.auto = auto;
   }
 
   public String sql() {
     return sql;
   }
 
-  public long statement() {
-    return statement;
+  /**
+   * Indicate whether this command is scheduled from {@link io.vertx.sqlclient.SqlClient#preparedQuery(String) one-shot preparedQuery} or {@link io.vertx.sqlclient.SqlConnection#prepare(String)}.
+   * This flag will tell if lifecycle of the statement will be controlled by the client or not.
+   *
+   * @return true if the command is a one-shot prepared query.
+   */
+  public boolean auto() {
```

Comment:
I think this should not be public because it is not needed by back-ends, this is an package protected internal data that is relevant only to the API and `SocketConnectionBase`, also the name should be more explicit like `shared` or `cacheable` that will tell whether it the prepared statement result can be shared or not.

Reply:
the rename is updated in https://github.com/eclipse-vertx/vertx-sql-client/pull/578/commits/9408a3b5b127a37cc74af466c55ce26d1d07ad89, changing the `public` access modifier will not be possible because they're in different packages.